compiler/wasm/src/compile.rs

```diff
@@ -148,8 +140,8 @@ impl PathToFileSourceMap {
 }
 
 pub enum CompileResult {
-    Contract { contract: ContractArtifact, debug: DebugArtifact },
-    Program { program: ProgramArtifact, debug: DebugArtifact },
+    Contract { contract: ContractArtifact, warnings: Vec<String> },
+    Program { program: ProgramArtifact, warnings: Vec<String> },
```

Comment:
I realise now that we're actually dropping all of the warnings from the `noirc_frontend`. Shall we drop these (warnings from `noirc_evaluator`) as well for now and then we can add a `warnings` field which contains both of these?

Reply:
On third thoughts, I'm fairly sure that the types as emitted from the wasm are now completely ignored. There's types in the TS code which are what are meaningful now.

Reply:
Ideally thought we'd still surface these types from the wasm into the TS. Seems like the lazy loading is throwing a spanner in the works however.

Reply:
See https://github.com/AztecProtocol/aztec-packages/pull/3989#issuecomment-1892720070